txpipe/twopoint.py

```diff
@@ -581,30 +577,31 @@ def calculate_pos_pos(self, data, i, j):
         n_i = cat_i.nobj
         n_rand_i = rancat_i.nobj if rancat_i is not None else 0
 
+        nn = treecorr.NNCorrelation(self.config)
+        rn = treecorr.NNCorrelation(self.config)
+        nr = treecorr.NNCorrelation(self.config)
+        rr = treecorr.NNCorrelation(self.config)
+        
         if i==j:
-            cat_j = cat_i
-            rancat_j = rancat_i
             n_j = n_i
             n_rand_j = n_rand_i
+            nn.process(cat_i)
+            nr.process(cat_i, rancat_i)
```

Comment:
In this case isn't nr the same as rn, so isn't this wasting time?

Reply:
I would assume so but it was already written like this before by somebody else, not sure if they had their reasons, maybe @empEvil knows?

Reply:
We can just do it for now, it's not a major thing, and we will need some changes for MPI treecorr anyway.

Reply:
Actually I think you're right, before it was written like this since there was not a specific option for when the catalogs were equal. Let me quickly fix it since we're here.